discovery/gossiper.go

```diff
@@ -1503,10 +1470,11 @@ func (d *AuthenticatedGossiper) addNode(msg *lnwire.NodeAnnouncement) error {
 func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 	nMsg *networkMsg) []networkMsg {
 
-	// isPremature *MUST* be called with the gossiper's lock held.
 	isPremature := func(chanID lnwire.ShortChannelID, delta uint32) bool {
 		// TODO(roasbeef) make height delta 6
 		//  * or configurable
+		d.Lock()
```

Comment:
ah okay, ignore my prior comment. i missed that _this_ method accesses `d.bestHeight`. it is still the case that we access `d.bestHeight` in the log statements so the locking should remain unchanged